### ISSUE TITLE
Add an option to print cwd to stdout on exit

### ISSUE THREAD

**Issue by WieeRd** [sxyazi/yazi#1960]
### `yazi --debug` output

```Shell
Yazi
    Version: 0.3.3 (Arch Linux 2024-09-05)
    Debug  : false
    OS     : linux-x86_64 (unix)

Ya
    Version: 0.3.3 (Arch Linux 2024-09-05)

Emulator
    Emulator.via_env: ("xterm-kitty", "")
    Emulator.via_csi: Ok(Kitty)
    Emulator.detect : Kitty

Adapter
    Adapter.matches: Kitty

Desktop
    XDG_SESSION_TYPE           : Some("x11")
    WAYLAND_DISPLAY            : None
    DISPLAY                    : Some(":0")
    SWAYSOCK                   : None
    HYPRLAND_INSTANCE_SIGNATURE: None
    WAYFIRE_SOCKET             : None

SSH
    shared.in_ssh_connection: false

WSL
    WSL: false

Variables
    SHELL              : Some("/usr/bin/nu")
    EDITOR             : Some("nvim")
    VISUAL             : Some("nvim")
    YAZI_FILE_ONE      : None
    YAZI_CONFIG_HOME   : None

Text Opener
    default: Some(Opener { run: "${EDITOR:-vi} \"$@\"", block: true, orphan: false, desc: "$EDITOR", for_: None, spread: true })
    block  : Some(Opener { run: "${EDITOR:-vi} \"$@\"", block: true, orphan: false, desc: "$EDITOR", for_: None, spread: true })

Multiplexers
    TMUX               : false
    tmux version       : tmux 3.5a
    ZELLIJ_SESSION_NAME: None
    Zellij version     : No such file or directory (os error 2)

Dependencies
    file             : 5.45
    ueberzugpp       : No such file or directory (os error 2)
    ffmpegthumbnailer: No such file or directory (os error 2)
    magick           : 7.1.1-41
    fzf              : 0.56.3
    fd               : 10.2.0
    rg               : 14.1.1
    chafa            : No such file or directory (os error 2)
    zoxide           : 0.9.6
    7z               : 17.05
    7zz              : No such file or directory (os error 2)
    jq               : 1.7.1
```

### Please describe the problem you're trying to solve

Yazi currently supports the `--cwd-file` flag to save the working directory on exit.

```bash
# `cd` to yazi's working directory on exit
function y() {
	local tmp="$(mktemp -t "yazi-cwd.XXXXXX")" cwd
	yazi "$@" --cwd-file="$tmp"
	if cwd="$(command cat -- "$tmp")" && [ -n "$cwd" ] && [ "$cwd" != "$PWD" ]; then
		builtin cd -- "$cwd"
	fi
	rm -f -- "$tmp"
}
```

While it works, that's quite a handful for such a simple task.
I've seen it done in a much simpler way from another TUI file manager, `xplr`.

### Would you be willing to contribute this feature?

- [X] Yes, I'll give it a shot

### Describe the solution you'd like

[Xplr](https://github.com/sayanarijit/xplr) has a `--print-pwd-as-result` flag, which, of course, prints the pwd on exit.
This allowed me to define a much simpler alias for changing the working directory on exit.

```bash
alias x='cd "$(xplr --print-pwd-as-result)"'
```

I'd be great to see this feature implemented in Yazi.

### Additional context

I've looked into relevant parts of Xplr's code to see how it works.

[`src/runner.rs # Runner::run() # L381-L383`
](https://github.com/sayanarijit/xplr/blob/dd731ebd3d08d321f887b5f36def7f4c419bd8f1/src/runner.rs#L381-L383)

```rust
PrintResultAndQuit => {
    result = if self.print_pwd_as_result {
        Ok(Some(app.pwd_str(self.delimiter)))
```

[`src/runner.rs # Runner::run() # L821-L830`](https://github.com/sayanarijit/xplr/blob/dd731ebd3d08d321f887b5f36def7f4c419bd8f1/src/runner.rs#L821-L830)

```rust
    terminal.clear()?;
    terminal.set_cursor(0, 0)?;
    execute!(terminal.backend_mut(), term::LeaveAlternateScreen)?;
    execute!(terminal.backend_mut(), event::DisableMouseCapture).unwrap_or_default();
    term::disable_raw_mode()?;
    terminal.show_cursor()?;
    
    fs::remove_dir_all(session_path)?;
    
    result
}
```

[`src/bin/xplr.rs # main() # L71-L73`
](https://github.com/sayanarijit/xplr/blob/dd731ebd3d08d321f887b5f36def7f4c419bd8f1/src/bin/xplr.rs#L71-L73)

```rust
match runner::from_cli(cli).and_then(|a| a.run()) {
    Ok(Some(out)) => {
        print!("{out}");
```

The TUI backend crate used by Xplr is `crossterm`. I do not know much about how TUIs work but looks like it is possible to leave the original stdout intact while rendering things on an alternative screen.

### Validations

- [X] I have searched the existing issues/discussions
- [X] The [latest nightly build](https://yazi-rs.github.io/docs/installation/#official-binaries) doesn't already have this feature

**Comment by sxyazi** [comment]
Duplicate of https://github.com/sxyazi/yazi/discussions/823#discussioncomment-8811630

**Comment by WieeRd** [comment]
Thanks! Haven't thought of utilizing `/dev/stdout`.
But wouldn't it better to add a dedicated flag, or at least support `-` as an alias to the stdout (e.g. `--cwd-file -`)?
`/dev/stdout` is \*nix specific, while Yazi is cross-platform.

**Comment by sxyazi** [comment]
That sounds great, too! 

Hopefully, in the end, we can use the new one to replace the existing `--cwd-file`, because the current shell wrapper relies on other external commands (like `cat` and `mktemp`). Some users might have aliased `bat` to `cat`, and some users might not have `mktemp` in their shell, so it's not always reliable. Do you want to give it a try?

Let's reopen it

**Comment by sxyazi** [comment]
Hmm it seems like the new one let's call it `--print-cwd`, might conflict with Yazi's `--local-events` and `--remote-events` because they all use `stdout`. 

In that case, it would be hard to distinguish which output is from `--print-cwd`. Maybe its significance isn't that great after all. What do you think?

**Comment by WieeRd** [comment]
Perhaps they can be moved to `stderr` - which is the intended channel for logs and real time reports, so it's kind of fitting. But that would be a breaking change to existing integrations using the events through stdout, so it has to be yet another opt-in option.

Also, since `--chooser-file` also exists, I think implementing a `-` alias to stdout will contribute to making the codebase more orthogonal and allow both `--cwd-file`, `--chooser-file` and potentially future features to use stdout more conveniently.

**Comment by sxyazi** [comment]
IMO the advantage of the existing `--cwd-file` and `--chooser-file` is that they can be used together with `--local-events` and `--remote-events`, where `--local-events` goes to stdout, while `--cwd-file` and `--chooser-file` go to files.

Also, `--cwd-file` and `--chooser-file` can be used together as long as you specify different file paths for each.

But the new `--print-cwd` and `--print-choice` cannot be used together with `--local-events` or `--remote-events`, nor can they be used together with each other. To me, this seems a bit redundant, especially considering that we already have `--cwd-file` and `--chooser-file`.

As for the issue you mentioned with `--cwd-file /dev/stdout`, this is indeed a cross-platform problem on Windows. However, Windows users can always specify an actual file path for it, so I don't think this will be a big issue.

> Perhaps they can be moved to `stderr`

Unfortunately, this is not possible - Yazi TUI is rendered in `stderr`

**Comment by WieeRd** [comment]
True, `/dev/stdout` already works just fine and also using temp files is not a deal breaker at all.

I was merely hoping to make my dotfiles cross-platform since I happen to be using a cross-platform shell which also happen to not support conditional aliases yet (Nushell). Very very niche and optional requirements that probably only applies to me lol.

**Comment by sxyazi** [comment]
Ok, let's close this issue then. 

Thanks for your suggestion for this feature anyway and your patience in discussing the specific details with me!
